src/app/parcels/getParcelsTableData.ts

```diff
@@ -43,30 +44,29 @@ export const processingDataToParcelsTableData = (
 
     for (let index = 0; index < processingData.length; index++) {
         const parcel = processingData[index];
-        const client = parcel.client!;
 
         parcelTableRows.push({
-            parcelId: parcel.parcel_id,
-            primaryKey: client.primary_key,
-            fullName: client.full_name,
-            familyCategory: familyCountToFamilyCategory(client.family.length),
-            addressPostcode: client.address_postcode,
-            phoneNumber: client.phone_number,
+            parcelId: parcel.parcel_id ?? "",
+            clientId: parcel.client_id ?? "",
+            fullName: parcel.client_full_name ?? "",
+            familyCategory: familyCountToFamilyCategory(parcel.family_count ?? 0),
+            addressPostcode: parcel.client_address_postcode ?? "",
+            phoneNumber: parcel.client_phone_number ?? "",
             deliveryCollection: {
-                collectionCentreName: parcel.collection_centre?.name ?? "-",
-                collectionCentreAcronym: parcel.collection_centre?.acronym ?? "-",
+                collectionCentreName: parcel.collection_centre_name ?? "-",
+                collectionCentreAcronym: parcel.collection_centre_acronym ?? "-",
                 congestionChargeApplies: congestionCharge[index].congestionCharge,
             },
             collectionDatetime: parcel.collection_datetime
                 ? new Date(parcel.collection_datetime)
                 : null,
             packingTimeLabel: datetimeToPackingTimeLabel(parcel.packing_datetime),
-            lastStatus: eventToLastStatus(parcel.events[0] ?? null),
+            lastStatus: processLastStatus(parcel),
             voucherNumber: parcel.voucher_number,
             packingDatetime: parcel.packing_datetime ? new Date(parcel.packing_datetime) : null,
             iconsColumn: {
-                flaggedForAttention: client.flagged_for_attention,
-                requiresFollowUpPhoneCall: client.signposting_call_required,
+                flaggedForAttention: parcel.client_flagged_for_attention ?? false,
+                requiresFollowUpPhoneCall: parcel.client_signposting_call_required ?? false,
```

Comment:
the changes in this file are mostly due to an annoying supabase limitation where it thinks any field in a view can be nullable. So when converting the response to be of type ParcelsTableRow[], we have to pass in a default value for every non-optional attribute of ParcelsTableRow.